graphistry/feature_utils.py

```diff
@@ -619,11 +682,19 @@ def fit_pipeline(
     columns = X.columns
     index = X.index
 
-    X = transformer.fit_transform(X)
-    if keep_n_decimals:
-        X = np.round(X, decimals=keep_n_decimals)  #  type: ignore  # noqa
-
-    return pd.DataFrame(X, columns=columns, index=index)
+    X_type = str(getmodule(X))
+    if 'cudf' not in X_type:
+        X = transformer.fit_transform(X)
+        if keep_n_decimals:
+            X = np.round(X, decimals=keep_n_decimals)  #  type: ignore  # noqa
+        X = pd.DataFrame(X, columns=columns, index=index)
+    else:
+        X = transformer.fit_transform(X.to_numpy())
```

Comment:
how do we know if the transformer is cpu vs gpu? it seems to always assume cpu here, but if X is cudf and transformer is gpu, can't we keep X on gpu?

Reply:
a sometimes-ok soln would be checking transformer for being from cuml or maybe cu_cat, but that seems pretty iffy

Reply:
wow this nearly gave me a heart attack -- good thoughts i will work with... but this is an artifact, no `.to_numpy` needed